cloudfour/cloudfour.com-patterns
# Chore: Refactor Button & Icon from "objects" to "components"

## Overview

This PR includes:
- moves Button and Icon source from `objects/` to `components/`
- renames Button and Icon class names so they are namespaced `c-` instead of `o-`
- updates any documentation referring to these patterns as "objects"
- changes Table heading from "Table Object" to "Table" (figured I could sneak this in per [PR feedback](https://github.com/cloudfour/cloudfour.com-patterns/pull/565#discussion_r397486212))

## Screenshots

### Button

<img width="1027" alt="Screen Shot 2020-03-24 at 2 43 57 PM" src="https://user-images.githubusercontent.com/459757/77479905-ed9e5c80-6ddd-11ea-9765-79e590267940.png">

### Icon

<img width="1027" alt="Screen Shot 2020-03-24 at 2 44 05 PM" src="https://user-images.githubusercontent.com/459757/77479906-ee36f300-6ddd-11ea-86b8-f309c1ed963c.png">


## Testing

1. Confirm the Button now exists under "Components" (e.g. `?path=/docs/components-button--elements`)
1. Confirm the Icon now exists under "Components" (e.g. `?path=/docs/components-icon--basic`)
1. From the **Design > Icons** page, confirm the "see the icon object" link correctly links to the new Icon URL

---

- Closes #558
